clients/ranchercli/download.go

```diff
@@ -0,0 +1,96 @@
+package ranchercli
+
+import (
+	"fmt"
+	osys "os"
+	"os/exec"
+	"strings"
+
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	Session *session.Session
+}
+
+func (c *Client) Create(name string, args ...string) error {
+	c.ExecuteCommand(name, args...)
+
+	return nil
+}
+
+func (c *Client) Exists(rancher, resourceType, name string) (bool, error) {
+	cmdStr := fmt.Sprintf("%s %s ls | grep %s", rancher, resourceType, name)
+	cmd := exec.Command("sh", "-c", cmdStr)
+	err := cmd.Run()
```

Comment:
I don't think this is the error we would want. Even if `name` isn't in the output of ls, the grep command will not exit with a nonzero code afaik. 

Instead I think you'd have to get the output of the `ls` or write a short bash if statement that will fail if the `name` isn't in the output of ls. maybe `if [[ -n $(%s %s ls | grep %s) ]]; then exit(0); else; exit(1); fi;` ?

Reply:
I actually tested this in the separate functions with bogus names  like `test` and `bogus` and it errored each time.

Reply:
hmm. mine doesn't error on a linux machine 🤔

Reply:
Met offline and got this working.